cmake/CompilerWarnings.cmake

```diff
@@ -35,6 +35,7 @@ function(set_target_warnings target)
             /wd4068 # disable warnings about unknown pragmas (e.g. #pragma GCC)
             /wd4505 # disable warnings about unused functions that might be platform-specific
             /wd4800 # disable warnings regarding implicit conversions to bool
+            /wd4275 # disable warnings about exporting non DLL-interface classes
```

Comment:
The alternative and maybe better way (?) is to not export the Exception and just have the constructor defined in the header, see https://stackoverflow.com/a/48453230/1034248 or https://stackoverflow.com/a/24512237/1034248

I'm okay with both ways. Ignoring the warning globally could theoretically hide some other issues, but not sure we'd really run into those.